### DISCUSSION TITLE
it's possible voice change to clone new voice with just one wav file or more ?

### DISCUSSION THREAD

**Discussion by lpscr** [p0p4k/pflowtts_pytorch#13] [Q&A] [answered]
hi @p0p4k thank you very much for this amazing work !! ,  
i just wonder if possible to use wave file for speaker , so for example i train a speaker and when i have good model
i can easy change with just  use wave file some like freevc if you know  , 

i see in interface in colab in this part of code

```
wav_files = glob.glob(path_waves+"*.wav") ## fill in the path to the LJSpeech-1.1 dataset

wav, sr = torchaudio.load(wav_files[0])
from pflow.data.text_mel_datamodule import mel_spectrogram
mel = mel_spectrogram(
            wav,
            1024,
            80,
            22050,
            256,
            1024,
            0,
            8000,
            center=False,
        )
```
and i wonder if possible , to use other voice  , because i try use another wave for voice speak , and the voice same always like i have train the basic speaker it's not change , 

if this possible be great ,

thank you

**Comment by p0p4k** [comment, marked as answer]
According to the paper's demo page, it should be possible when we train the base model on many speakers and use a slightly larger model. https://pflow-demo.github.io/projects/pflow/#:~:text=The%20samples%20below,P%2DFlow%20(v1.5)

**Comment by lpscr** [comment]
hi @p0p4k thank you very much for quick reply , i try to train with multi speaker to see how it's going  
just to know the dataset  i think work like vits2 same format in 

`wav file|speker_id|text`

i run same file like i train the single speaker and i change yaml,file to vctk.yaml and change inside the path files numbr speaker etc ..

`python pflow/train.py experiment=vctk.yaml`

this be work ? can i use to train like this ?

when i train complete with multi speaker and i want to run ,  i change the wav  like i say up or need extra code for this to change the voice ?

thank you

**Comment by p0p4k** [comment]
Same code mostly. You might have to calculate stats before training for your dataset and replace numbers in the yaml file.

**Comment by lpscr** [comment]
@p0p4k great thank you so much for your help, i love your work ! i try train with multi speaker in next days because right now i train with single speaker and i am 150 k about if possible to use single train model when i finish, and use to fine tune a multi speaker or need start from scratch again ? 

here how look like this i train it's ok so far ? single speaker
![p1](https://github.com/p0p4k/pflowtts_pytorch/assets/147736764/72b5710e-7ca4-416d-b93f-d0f250eccb8c)


![p2](https://github.com/p0p4k/pflowtts_pytorch/assets/147736764/06a71270-3417-4a16-95fc-d8640b50c88e)

i notice there is empty space i have mark with orange paint, this ok , like i see in fron page your it's full fit
![p3](https://github.com/p0p4k/pflowtts_pytorch/assets/147736764/169a3095-f969-4dda-9fa2-123ffd395535)

**Reply by p0p4k** [reply]
I moved this to discussion as it is not an issue anymore.
You can do transfer learning from single speaker, by changing `checkpointpath` in `train.yaml` under `configs`.
The gap is not a problem, because in the model bunch of wavs come in as a matrix and they all have different lengths, so the blank space you see is called `padding`. You might want to read a bit more about these things before going deeper and look at the code carefully to understand it more.

**Reply by lpscr** [reply]
@p0p4k Great thank you again for your  help , i try see in next day the multi speaker train
